### ISSUE TITLE
Add custom ETA calculation

### ISSUE THREAD

**Issue by alexheretic** [alexheretic/ab-av1#174]
Current eta provided by upstream _indicatif_ has [issues with slow ticks](https://github.com/console-rs/indicatif/issues/580) of some encode jobs. It also struggles with switching between different rates of encoding and VMAF part of a single job.

A custom eta calculation tailored for ab-av1 usage may do better.

**Comment by alexheretic** [comment]
One workaround could be to increase granularity from using encoding progress `seconds / total_seconds`  to `µs / total_µs` to provide the  upstream algorithm more regular progress.

**Comment by alexheretic** [comment]
#175 improves some scenarios but I can still see it eta rocketing up to years when it should be minutes.

**Comment by zedstate** [comment]
Why not base the ETA on frames per second?

**Comment by alexheretic** [comment]
> Why not base the ETA on frames per second?

Indeed. Currently we use upstream eta calc based on progress bar updates. So yes replacing that with a custom calculation we can base it on ffmpeg reported rate. We can also account for vmaf rate & encoding rate in crf-searches.
